File: app/src/main/kotlin/net/tigr/soulcalc/domain/model/Result.kt
/*
 * SPDX-License-Identifier: MIT
 * Copyright (c) 2026 Albert Zenkoff
 */

package net.tigr.soulcalc.domain.model

/**
 * Represents the result of evaluating a single line.
 */
sealed class Result {
    /** Successfully evaluated to a numeric value. */
    data class Success(val value: Double) : Result()

    /** Evaluation failed with an error. */
    data class Error(val message: String) : Result()

    /** Line is empty or a comment - no result to display. */
    data object Empty : Result()
}

File: ios/SoulCalc/Domain/Model/Result.swift
// SPDX-License-Identifier: MIT
// Copyright (c) 2026 Albert Zenkoff

import Foundation

/// Represents the result of evaluating a single line.
enum Result {
    /// Successfully evaluated to a numeric value.
    case success(Double)

    /// Evaluation failed with an error.
    case error(String)

    /// Line is empty or a comment - no result to display.
    case empty
}
